### PR TITLE
Dashboards page update

### PR DESCRIPTION
**Pull request by Woafflation**
## Changes

Base PR for the dashboards update

1. Updated NavBar of Dashboards page
 1.1 Updated Nav styles and components, removed nav tooltips and added new tabs
 1.2 Updated nav scrolling hook, moved all logic into independent hook, added scroll observer logic
 1.3 Fixed scrolling to item
2. Added new header for Dashboards
 2.2 Updated Share via media, added new icons and links
 2.3 Updated title and description
3. Updated blocks and their styles
4. Social tool update
 4.1 Added 'Top 10 Hourly Trends', refactored it and updated styles
 4.2 Updated Search component, refactored it and fixed styles
 4.3 Added 'Social dominance sum' tooltip for table
 4.4 Added date selector for trends table
 4.5 Updated Share table view button
 4.6 Added 'Average social volume, 7d' column
 4.7 Added 'Popular Mid-Term Trends', refactored it and updated styles
5. Added 'Ethereum Staking Pools', refactored it and updated styles

## Notion's card

https://www.notion.so/santiment/Dashboard-style-updates-Social-tool-combine-2924c81d91c349249fd12fec4abe5ccf

## Checklist

- [x] I've performed a self-review, followed all rules from [Frontend style guide](https://www.notion.so/santiment/Front-end-style-guide-81750096b38c4bea9a29b14fd4ab8667)
- [x] If I make changes to another person's module, I've asked how to use it or request a review
- [x] I've updated the [documentation](https://github.com/santiment/academy), if necessary (Keyboard shortcuts, Account settings)
- [x] I've checked night mode, mobile & tablet screens (if have changes in UI)
- [x] I've added tests (if necessary)

## Screenshots or GIFs

Снимок экрана 2022-10-17 в 03 37 59
Снимок экрана 2022-10-17 в 03 38 03
Снимок экрана 2022-10-17 в 03 38 08
Снимок экрана 2022-10-17 в 03 38 12
Снимок экрана 2022-10-17 в 03 38 21
Снимок экрана 2022-10-17 в 03 38 49
Снимок экрана 2022-10-17 в 03 38 58
Снимок экрана 2022-10-17 в 03 39 29
Снимок экрана 2022-10-17 в 03 39 39

### CODE REVIEW

File: src/pages/Dashboards/Nav/NavItem/NavItem.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          <SubItem
+            key={subItem.key}
+            activeSub={activeSub}
+            subItem={subItem}
+            scrollToSubItem={scrollToSubItem}
+          />
+        ))}
+      </ul>
+    )
+  }
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/Nav/NavItem/NavItem.js:41]: Conditionally render element on the `LOC 56` -> `{activeSub && <ul className ....}`

File: src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/Columns/TopTrendsColumns.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      title: Column.SOCIAL_VOLUME,
+      render: (trend, { words }) => <AverageSocialVolume trend={trend} words={words} />,
+    },
+    {
+      title: Column.CONNECTED_WORDS,
+      className: styles.cloud,
+      render: (trend, { words }) => <ConnectedWords trend={trend} words={words} />,
+    },
+  ]),
+)
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/Columns/TopTrendsColumns.js:95]: Reuse components from `src/ducks/TrendsTable/columns.js`

File: src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/TopTrendsTable.module.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  th:first-of-type {
+    padding-left: 24px;
+  }
+  th:last-of-type {
+    padding-right: 24px;
+  }
+}
+
+.word {
+}
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/TopTrendsTable.module.scss:36]: Empty class

File: src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/TopTrendsTable.module.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+.table {
+  min-height: 350px;
+  border-bottom-left-radius: 8px;
+  border-bottom-right-radius: 8px;
+
+  td {
+    padding: 8px 12px;
+  }
+
+  td:first-of-type {
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/TopTrendsTable.module.scss:12]: `&:first-of-type` inside `td`

File: src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/TopTrendsTable.module.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    color: var(--waterloo);
+    padding: 8px 12px;
+  }
+
+  th:first-of-type {
+    padding-left: 24px;
+  }
+  th:last-of-type {
+    padding-right: 24px;
+  }
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/TopTrendsTable.module.scss:32]: Put `:first-of-type` and `:last-of-type` inside `th` selector and use `&`

File: src/pages/Dashboards/dashboards/SocialTool/TrendsExploreInput/TrendsExploreInput.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        className={cx(styles.input, 'input fluid', showButton && styles.inputWithButton)}
+        value={topic}
+        onChange={handleTopicChange}
+      />
+      {showButton && btn}
+    </form>
+  )
+}
+
+export default connect(null, gotoExplore)(TrendsExploreInput)
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/SocialTool/TrendsExploreInput/TrendsExploreInput.js:44]: Duplicate of the `src/components/Trends/TrendsExploreInput/TrendsExploreInput.js`

File: src/pages/Dashboards/dashboards/SocialTool/TrendsExploreInput/TrendsExploreInput.module.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+.inputWithButton {
+  padding: 9px 94px 9px 13px;
+}
+
+.explore {
+  position: absolute;
+  right: 4px;
+  top: 4px;
+}
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/SocialTool/TrendsExploreInput/TrendsExploreInput.module.scss:17]: Duplicate of the `src/components/Trends/TrendsExploreInput/TrendsExploreInput.module.scss`

File: src/App.js [hunk 8]
```diff
@@ -489,6 +490,10 @@ export const App = ({
               path={PATHS.DASHBOARDS}
               render={(props) => <LoadableDashboardsPage {...props} />}
             />
+            <Route
+              path={`${PATHS.DASHBOARDS}-old`}
+              render={(props) => <LoadableOldDashboardsPage isDesktop={isDesktop} {...props} />}
+            />
```
Comment (DmitriVanGuard) [inline src/App.js:496]: What is the reason for keeping the old route for production? It will not be maintained

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 8 above)
Comment (Woafflation) [reply src/App.js:496]: It's not the final PR, so there are some duplicates and old route
Also updating old components to new styles

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]
Change styles and render of the existing Nav component instead of creating a new completely. Same is true for smooth scrolling

File: src/pages/Dashboards/dashboards/shared/ShareMedia/ShareMedia.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+          className={cx(styles.btn, 'btn row hv-center')}
+        >
+          <Icon />
+        </a>
+      ))}
+    </div>
+  )
+}
+
+export default ShareMedia
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/shared/ShareMedia/ShareMedia.js:52]: Duplicate of the `src/components/Share/medias/ShareMedias.js`

File: src/pages/Dashboards/dashboards/SocialTool/TrendsExploreInput/TrendsExploreSuggestions.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+          {trend}
+          {index !== SUGGESTIONS.length - 1 && ','}
+        </Link>
+      ))}
+    </p>
+    <HelpPopupTrends className={styles.tooltip} triggerClassName={styles.tooltipIcon} />
+  </div>
+)
+
+export default TrendsExploreSuggestions
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/SocialTool/TrendsExploreInput/TrendsExploreSuggestions.js:41]: Duplicate of the `src/components/Trends/Search/Suggestions.js`

File: src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/TopTrendsTable.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+      itemKeyProperty='word'
+      itemProps={{ words }}
+      minRows={10}
+      isLoading={isLoading}
+      onRowClick={onRowClick}
+    />
+  )
+}
+
+export default TopTrendsTable
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/TopTrendsTable.js:36]: Duplicate of the `src/ducks/TrendsTable/index.js`

File: src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/Columns/TopTrendsColumns.js [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/SocialTool/TopTrends/Table/Columns/TopTrendsColumns.js:105]: Duplicate of the `src/ducks/TrendsTable/columns.js`

File: src/pages/Dashboards/dashboards/EthStakingPools/EthStakingPools.js [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+      title='Ethereum Staking Pools'
+      description='Information all about staking metrics and statistics for the new Ethereum 2.0.'
+    />
+    <main className={cx(dashboardsStyles.content, 'column')}>
+      <div id='eth_validators'>
+        <h4 className='h4 txt-b mrg-xxl mrg--b'>Number of Validators</h4>
+        <div>
+          <EthStakingPools />
+        </div>
+      </div>
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/EthStakingPools/EthStakingPools.js:21]: Create `Section` component with `id`, `title` and `children` props. Reuse it in all dashboards

File: src/pages/Dashboards/dashboards/shared/ProCheck/ProCheck.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import React, { Fragment, useMemo } from 'react'
+import CheckProPaywall from '../../../../../ducks/Stablecoins/CheckProPaywall'
+
+const ProCheck = ({ isPaywalActive = false, children }) => {
+  const El = useMemo(() => {
+    return isPaywalActive ? CheckProPaywall : Fragment
+  }, [isPaywalActive])
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/shared/ProCheck/ProCheck.js:7]: `useMemo` is redundant

File: src/pages/Dashboards/dashboards/shared/Info/Info.js [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+const Info = ({ title, description }) => (
+  <article className={cx(styles.wrapper, 'row justify')}>
+    <div className={cx(styles.info)}>
+      <h4 className='h4 txt-b mrg-m mrg--b'>{title}</h4>
+      <p>{description}</p>
+    </div>
+    <div className={cx(styles.share, 'column')}>
+      <p className='nowrap txt-m txt-right c-waterloo mrg-m mrg--b'>Share on social media</p>
+      <ShareMedia text={description} />
+    </div>
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/shared/Info/Info.js:15]: Copy of the `src/components/Share/medias/ShareMedias.js`

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/pages/Dashboards/dashboards/shared/Share/Share.js [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+    if (clearTimerRef.current) clearTimerRef.current()
+
+    getShortShareLink(window.location.pathname).then((url) => {
+      const node = document.getElementById(id)
+      const clb = () => node && (node.ariaLabel = 'Copy link')
+
+      if (node) node.ariaLabel = 'Copied!'
+      clearTimerRef.current = copy(url, clb)
+    })
+  }
```
Comment (DmitriVanGuard) [inline src/pages/Dashboards/dashboards/shared/Share/Share.js:29]: 1. Short link is not needed;
2. `event.currentTarget`
3. Clear timer on unmount

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

**Review by DmitriVanGuard — APPROVED** [review]

**Pull request merged**